designcompose/src/main/java/com/android/designcompose/FrameRender.kt

```diff
@@ -370,12 +370,38 @@ internal fun ContentDrawScope.render(
                                 progressBarData.discrete,
                                 progressBarData.discreteValue
                             )
-                        if (progressBarData.vertical) {
-                            frameSize = Size(size.width, progress)
-                            val topOffset = style.top.pointsAsDp().value
-                            overrideTransform = androidx.compose.ui.graphics.Matrix()
-                            overrideTransform.translate(0F, -topOffset + end - progress)
-                        } else frameSize = Size(progress, size.height)
+                        if (progressBarData.moveOnly) {
```

Comment:
The code for if and else are very simmilar, and it's indented seven times. I have a few suggestions:

#1, refactor the code in the if and else blocks into a function.
#2, move the contents of the whole "is MeterData.progressBarData" block into a separate function. 
#3, can you add a unit test for the progress bar? Not a visual test, just a test of the math.

Reply:
This has been heavily refactored.